### PR TITLE
Unified extended test binary

### PR DESCRIPTION
**Pull request by marun**
The core and networking extended tests were previously built into
separate binaries due to the differing test setup required.  This change
includes all tests in the same binary to simplify usage by downstream
consumers like QE.

This PR depends on #7875 (merge pending) to ensure that networking 
tests will skip instead of failing when isolation and multinode are not 
available which should allow the core suite to attempt to execute 
networking tests without fear of a failed test run.

cc: @timothysc @jayunit100 @smarterclayton @openshift/networking

### CODE REVIEW

File: test/extended/networking.sh [hunk 1]
```diff
@@ -7,7 +7,7 @@ set -o errexit
 set -o nounset
 set -o pipefail
 
-if [[ "${OPENSHIFT_QUIET_OUTPUT:-false}" != "true" ]]; then
+if [[ "${OPENSHIFT_QUIET_OUTPUT:-true}" != "true" ]]; then
```
Comment (jayunit100) [inline test/extended/networking.sh]: why is this changing?

Comment (jayunit100) [inline test/extended/networking.sh]: NVM, i saw this depends on the other one pending :).

**Comment by danwinship** [comment]
- dropped "Make extended networking tests quiet(er) by default" (feel free to do something with it in your branch)
- dropped providerIs()
- (belatedly) renamed checkService() to checkServiceConnectivity()

**Comment by marun** [comment]
[testonlyextended][extended:all]

**Comment by marun** [comment]
[testonlyextended][extended:core]

**Comment by knobunc** [comment]
LGTM.

**Comment by smarterclayton** [comment]
[merge]

**Comment by marun** [comment]
Please hold off merging this just yet.  I haven't seen a successful run with extended:all, going to try extended:core and extended:networking separately to make sure they can pass.

**Comment by marun** [comment]
[testonlyextended][extended:networking]

**Comment by marun** [comment]
[testonlyextended][extended:core]

**Comment by marun** [comment]
The failures in the core extended suite don't appear to have anything to do with this PR.  Failures I would worry about would be origin networking tests running in the core suite and failing, but I see no evidence of that (all are passing or skipping).

So, please merge.

**Comment by marun** [comment]
[testonlyextended][extended:networking]

**Comment by smarterclayton** [comment]
[merge]

**Pull request merged**